src/main/java/commands/ToDoCommand.java

```diff
@@ -0,0 +1,46 @@
+package commands;
+
+import java.io.IOException;
+
+import cook.Storage;
+import cook.TaskList;
+import cook.Ui;
+import exceptions.InvalidInputException;
+import tasks.ToDo;
+
+/**
+ * ToDoCommand class to process ToDo commands.
+ */
+public class ToDoCommand extends Command {
+    protected String description;
+
+    /**
+     * Constructor for ToDoCommand class.
+     */
+    public ToDoCommand(String description) throws InvalidInputException {
+        super("todo");
+        this.description = description;
+    }
+
+    /**
+     * Adds ToDo task and saves file.
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
```

Comment:
For this execute method that is common for ToDoCommand, EventCommand and DeadlineCommand, perhaps you could consider putting it in Command class instead so that you don't have to repeat yourself. But great job on implementing this!

Reply:
Ah! The main reason I separated them out was because each command would need to create a specific instance of a different class.
``` java
// Create ToDo instance
new Todo(this.description)
// Create Deadline instance
new Deadline(this.description, this.by)
// Create Event instance
new Event(this.description, this.from, this.to)
```
I don't know of a clean way to combine all of this into a single method in the command class since it only contains the description attribute. Even if I did, the method would be inherited by the other commands which I don't really want to happen since they don't even make use of it.

The other option I thought of was to create another abstract class inheriting from Command, which all task creating commands will inherit from. This will reduce code duplication by containing it within the new abstract class, but I'm unsure of the feasibility of calling a super method within its overridden method in Java. Do let me know if you have other thoughts on this.